lib/internal/test_runner/test.js

```diff
@@ -172,10 +172,13 @@ class Test extends AsyncResource {
 
   constructor(options) {
     super('Test');
-
-    let { fn, name, parent, skip } = options;
+    let { fn, name, parent, skip, testNamePatterns = [] } = options;
```

Comment:
I think we should use `null` or `undefined` instead of creating a throw away array for every test (the test runner common case is **not** using `testNamePatterns`).